doc/architecture/decisions/0004-upgrade-testing.md

```diff
@@ -0,0 +1,40 @@
+# 3. upgrade testing methodology when releasing
+
+Date: 2020-06-25
+
+## Status
+
+Accepted
+
+## Context
+
+One of the goals of the KubeCF project is to offer a smooth experience when it comes to upgrading Cloud Foundry deployments.
+That means any user operating a KubeCF deployment should:
+
+- be able to easily identify which versions he can upgrade to
+- upgrade using documented procedures
+- have confidence that the exact version upgrade he's doing has been tested by the KubeCF team 
+
+## Decision
+
+- not allowed to have minor releases for version X.y.z if X+1 has been released (only patch releases)
+- for a version X.y.z, the upgrade path must be tested for:
+  - the previous X.*.* release
+  - version max(X-1.*.*)
+
+For downstream projects, assuming they don't release as often as KubeCF, the following rules should be in place:
+
+- there should be a release for each major KubeCF release
+- for a version X.y.z, the upgrade path must be tested for:
+  - the previous X.*.* release
+- if a version X.y.z cannot be upgraded from X.y-1.t, a new version X.y-1.t+1 must be created from a known upgradeable path in KubeCF.
+
+![diagram](https://docs.google.com/drawings/d/e/2PACX-1vSK_9XqNiLbpzrQGnJ9BISSQq8DKTeE3yDjszyJfC7BdPuABO0QbAMMZruEoMnTFwhhtzCEGeXowqmh/pub?w=1037&h=918)
```

Comment:
Nitpicking:
We are going to remove the 'v' out of our version strings, to comply with semver. May we worth it to remove it from the diagram too. Aso, maybe it is good to have the diagram as an asset file of the repo.

Reply:
I don't understand this graph:
- What's all the coloured boxes?
- Why is there a disconnected graph on the right?
- Why is v3.2.1 not upgradable from v2.2.0?

Reply:
There is also the direction of the arrows. Right now they seem to point from version to predecessor to upgrade from ?
With upgrades I would have expected arrows from version to descendant to upgrade to.

Reply:
Removed the v, added titles to the 2 graphs. Hopefully the colors make sense now @mook-as 
I changed the direction of the arrows too, I agree it's more intuitive this way @andreas-kupries

Reply:
I _think_ the colours basically mean "downstream 2.0.0 packages KubeCF 1.0.0", etc? If that's the case, it would be much better to have explicit text saying so (a dashed arrow pointing from KubeCF 1.0.0 to 2.0.0 with a caption, perhaps).

It might also be a good idea to embed text-representation of the graph, instead of linking to a Google doc that not everybody can edit (esp. for forks that want to PR changes).  Perhaps something like [this](https://github.com/TLmaK0/gravizo) would help.

Reply:
I agree with the caption, I end up making my own assumptions here when reading the chart.